WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationDeleteEventProcessor.swift

```diff
@@ -32,9 +35,56 @@ protocol ConversationDeleteEventProcessorProtocol {
 
 struct ConversationDeleteEventProcessor: ConversationDeleteEventProcessorProtocol {
 
-    func processEvent(_: ConversationDeleteEvent) async throws {
-        // TODO: [WPB-10167]
-        assertionFailure("not implemented yet")
+    enum Error: Swift.Error {
+        case failedToDeleteMLSConversation(Swift.Error)
+    }
+
+    let context: NSManagedObjectContext
+    let repository: any ConversationRepositoryProtocol
+    let mlsService: any MLSServiceInterface
+
+    func processEvent(_ event: ConversationDeleteEvent) async throws {
+        let id = event.conversationID.uuid
+        let domain = event.conversationID.domain
+
+        let conversation = await repository.fetchConversation(
+            with: id,
+            domain: domain
+        )
+
+        guard let conversation else {
+            return WireLogger.eventProcessing.warn(
+                "Cannot delete a conversation that doesn't exist locally: \(id.safeForLoggingDescription)"
+            )
+        }
+
+        if conversation.messageProtocol == .mls {
+            do {
+                try await wipeMLSGroup(conversation: conversation)
+            } catch {
+                throw Error.failedToDeleteMLSConversation(error)
+            }
+        }
+
+        await context.perform {
+            conversation.isDeletedRemotely = true
+        }
```

Comment:
I would expect this to be in a method `conversationRepository.deleteConversation(id: ...)`. Any particular reason why it's done here?

Reply:
fixed in https://github.com/wireapp/wire-ios/pull/2034/commits/2173b091d0d61b128c85d86785ff058230b8ea09